### PR TITLE
Feature/layer-group

### PR DESCRIPTION
**Pull request by krishnaglodha**
Added functions to get the layer group, which are optionally dependent on the workspace name.

### CODE REVIEW

**Comment by krishnaglodha** [comment]
Extra models for `creating new layer group` has been added in this PR. They will be used in upcoming features.

File: src/geoserverx/models/layer_group.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    name: str = ...
+    href: str = ...
+
+class LayerGroupList(BaseModel):
+    layerGroup: List[LayerGroupElement] = ...
+
+class LayerGroupsModel(BaseModel):
+    layerGroups: Union[LayerGroupList, Literal[""]]
+
+class PublishedDict(BaseModel):
```
Comment (francbartoli) [inline src/geoserverx/models/layer_group.py]: Does it make sense to refer to a dictionary in the class name?

File: src/geoserverx/models/layer_group.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    type: str = Field(..., alias="@type")
+    name: str = ...
+    href: str = ...
+
+class PublishablesDict(BaseModel):
```
Comment (francbartoli) [inline src/geoserverx/models/layer_group.py]: same as above

File: src/geoserverx/models/layer_group.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    published: PublishedDict = ...
+
+class StyleDict(BaseModel):
```
Comment (francbartoli) [inline src/geoserverx/models/layer_group.py]: same as above

File: src/geoserverx/models/layer_group.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    name: str = ...
+    href: str = ...
+
+class StylesDict(BaseModel):
```
Comment (francbartoli) [inline src/geoserverx/models/layer_group.py]: Same as above

File: src/geoserverx/models/layer_group.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    style: StyleDict = ...
+
+class BoundsDict(BaseModel):
```
Comment (francbartoli) [inline src/geoserverx/models/layer_group.py]: Same as above

File: tests/test_models.py [hunk 6]
```diff
@@ -31,7 +31,7 @@
     NewWorkspaceInfo,
     SingleWorkspace,
 )
-
+from geoserverx.models.layer_group import LayerGroupsModel
 
 # Testing DataStoreInBulk
 def test_DataStoreInBulk_connection(good_datastore_in_bulk_connection):
```
Comment (francbartoli) [inline tests/test_models.py]: function name lowercase

File: tests/test_models.py [hunk 7]
```diff
@@ -264,3 +264,25 @@ def test_NewWorkspace_connection(good_new_workspace_connection):
 def test_NewWorkspace_failure(bad_new_workspace_connection):
```
Comment (francbartoli) [inline tests/test_models.py]: Same as above

File: tests/test_models.py [hunk 8]
```diff
@@ -264,3 +264,25 @@ def test_NewWorkspace_connection(good_new_workspace_connection):
...
     with pytest.raises(ValidationError):
         ds_connection = NewWorkspace(**bad_new_workspace_connection)
+
+
+# Testing WorkspaceModel
+def test_WorkspaceModel_connection(good_workspace_model_connection):
```
Comment (francbartoli) [inline tests/test_models.py]: Same as above

File: tests/test_models.py [hunk 9]
```diff
@@ -264,3 +264,25 @@ def test_NewWorkspace_connection(good_new_workspace_connection):
...
+    ds_connection = WorkspaceModel(**good_workspace_model_connection)
+    assert ds_connection.workspace.name == "pydad"
+
+
+def test_WorkspaceModel_failure(bad_workspace_model_connection):
```
Comment (francbartoli) [inline tests/test_models.py]: Same as above

File: tests/test_models.py [hunk 10]
```diff
@@ -264,3 +264,25 @@ def test_NewWorkspace_connection(good_new_workspace_connection):
...
+    with pytest.raises(ValidationError):
+        ds_connection = WorkspaceModel(**bad_workspace_model_connection)
+
+
+
+# Testing LayerGroupsModel
+def test_LayerGroupsModel_connection(good_layer_groups_connection):
```
Comment (francbartoli) [inline tests/test_models.py]: Same as above

File: tests/test_models.py [hunk 11]
```diff
@@ -264,3 +264,25 @@ def test_NewWorkspace_connection(good_new_workspace_connection):
...
+    ds_connection = LayerGroupsModel(**good_layer_groups_connection)
+    assert ds_connection.layerGroups.layerGroup[0].name == "tg"
+
+def test_LayerGroupsModel_failure(bad_layer_groups_connection):
```
Comment (francbartoli) [inline tests/test_models.py]: Same as above

**Review by francbartoli — CHANGES_REQUESTED** [review]

**Comment by francbartoli** [comment]
Can you rebase and fix conflicts as per last changes in develop?

File: tests/test_models.py [hunk 12]
```diff
@@ -261,26 +261,4 @@ def test_newworkspace_connection(good_new_workspace_connection):
...
-        ds_connection = WorkspaceModel(**bad_workspace_model_connection)
-
-
-
-# Testing LayerGroupsModel
-def test_layergroupsmodel_connection(good_layer_groups_connection):
-    ds_connection = LayerGroupsModel(**good_layer_groups_connection)
-    assert ds_connection.layerGroups.layerGroup[0].name == "tg"
-
-def test_layergroupsmodel_failure(bad_layer_groups_connection):
```
Comment (krishnaglodha) [inline tests/test_models.py]: need to add back these tests again

**Pull request merged**